src/renderers/WebGLRenderer.js

```diff
@@ -902,15 +905,17 @@ function WebGLRenderer( parameters ) {
 
 		scene.traverse( function ( object ) {
 
-			let material = object.material;
+			objects.update( object ); // Upload geometry attributes
```

Comment:
This change is problematic since `WebGLObjects.update()` should only be called for instance of `Mesh`, `Points`, `Line` and `Sprite`. So objects that have a geometry.

Reply:
How about completely removing the changes related to `WebGLRenderer.compile()`. It seems they do not belong to this PR.

Reply:
Unfortunately it does belong to this PR, and that took me ~4h to figure it out. Let me try to summarize...

Basically, when we call `new WebGLCubeRenderTarget().fromEquirectangularTexture()` within a render, we're doing a nested render() which changes the state of the parent render() and we end up with a black render.

So I tried to change `fromEquirectangularTexture()` to be more like `WebGLShadowMap` which doesn't call `render()` but `renderBufferDirect()`. `renderBufferDirect()` depends on geometry attributes having been uploaded in order to work, and that's what happens in `objects.update( object )`.

Until I added this line I was just getting a black screen with no warnings anywhere (debugging WebGL is fun!).

Anyway, I didn't know if this was the right thing to do. Now I have to check what's the render flow when we render mirrors and see what approach to go with.

Reply:
Okay, this dependency is indeed not obvious^^.

Well, if there is no other solution you could still do:

```js
if ( object.geometry ) objects.update( object );
```
in order to fix `webgl_materials_compile`.

Reply:
Ah! Interesting. I'll give it a try tomorrow 🙏

Reply:
Done! 🤞

Reply:
Only one broken example left! Getting there 👍